src/System.Net.Security/src/System/Net/Security/SslConnectionInfo.Linux.cs

```diff
@@ -0,0 +1,87 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Win32.SafeHandles;
+using System.Security.Authentication;
+
+namespace System.Net.Security
+{
+    internal partial class SslConnectionInfo
+    {
+        public SslConnectionInfo(SafeSslHandle sslContext)
+        {
+            Protocol = (int)MapProtocolVersion(Interop.Ssl.SslGetVersion(sslContext));
+
+            MapCipherSuite(SslGetCurrentCipherSuite(sslContext));
```

Comment:
We should be consistent: it's strange to have one of these `Map...` methods update state and the other not.

Reply:
this was existing code I moved to different place so that we can use it on both Linux and OSX - also this maps 1 value into 6 so don't see any clean way to write that except for copy & paste